### PR TITLE
COM-3516 Show root account when account is not part of an account hie…

### PR DESCRIPTION
**Pull request by ghost**
…rarchy

### CODE REVIEW

**Comment by kibo-sonalsatpute** [comment]
Just refactored your changes, not tested. One thing I am not sure about the second parameter of getHierarchy(currentAccount, **self**); method see if there is a better way to get hold of `apiGet` method.
 
```

      initialize: function () {
            var self = this;
            self.initApiModel(); 
            var accountId = require.mozuData('user').accountId;
            var b2bAccount = new B2BAccountModels.b2bAccount({ id: accountId });
            b2bAccount.apiGet().then(function (response) {
         
                var hierarchy = getHierarchy( response.data, self);
                
                self.set("hierarchy", hierarchy);
                self.syncApiModel();
                self.trigger("render");
            });
            
            self.set("isUserAdmin", self.isUserAdmin());
        },
        getHierarchy : function(currentAccount, parent) {
            var self = parent;
            
            if (currentAccount && currentAccount.rootAccountId) {
                var hierarchy;
                self.apiGet({ id: accountId }).then(function (response) {
                    hierarchy = self.processAccountHierarchy(response.data)
                });
                return hierarchy;
            }
           
            var data = {
                accounts: [currentAccount],
                hierarchy: { id: accountId }
            };

            return self.processAccountHierarchy(data)
        },
```

File: scripts/modules/b2b-account/account-hierarchy.js [hunk 1]
```diff
@@ -122,17 +122,29 @@ define(["modules/jquery-mozu", 'modules/api', "underscore", "hyprlive", "modules
...
             self.set("isUserAdmin", self.isUserAdmin());
         },
+        getHierarchy: function (self, currentAccount) {
+            var accountId = require.mozuData('user').accountId;
+            if (currentAccount && currentAccount.rootAccountId) {
+                self.apiGet({ id: accountId }).then(function (response) {
+                    return self.processAccountHierarchy(response.data);
+                });
+            }
+            else {
```
Comment (kibo-sonalsatpute) [inline scripts/modules/b2b-account/account-hierarchy.js]: do we really need else block?

**Comment by ghost** [comment]
@kibo-sonalsatpute  the code you suggested doesn't work because we do the asynchronous operation and it fits synchronously but I still reworked a certain logic. Please take a look

**Review by kibo-sonalsatpute — APPROVED** [review]
👍  changes look good to me.

**Review by kibo-raghu — APPROVED** [review]
Approved

**Pull request merged**